tokio-trace-tower-http/src/lib.rs

```diff
@@ -77,14 +100,19 @@ where
     type Item = InstrumentedHttpService<T::Item>;
     type Error = T::Error;
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        let span2 = self.span.clone();
-        let span = &mut self.span;
-        let inner = &mut self.inner;
-        span.enter(move || {
-            inner
+        if let Some(span) = &mut self.span {
+            let span2 = span.clone();
+            let inner = &mut self.inner;
+            span.enter(move || {
+                inner
+                    .poll()
+                    .map(|ready| ready.map(|svc| InstrumentedHttpService::with_span(svc, span2)))
```

Comment:
It occurs to me that the separate `if`/`else` branches wouldn't be necessary if there was an `InstrumentedHttpService` constructor taking an `Option<Span>`...just a thought.